src/components/HeartbeatBar.vue

```diff
@@ -60,6 +65,14 @@ export default {
     },
     computed: {
 
+        /**
+         * Normalized heartbeatBarDays as a number
+         * @returns {number} Number of days for heartbeat bar
+         */
+        normalizedHeartbeatBarDays() {
+            return Math.max(0, Math.min(365, Math.floor(this.heartbeatBarDays || 0)));
+        },
+
```

Comment:
going outside the `[0,365]` range on the frontend should not be possible on the frontend without tampering with it.

Given the database check constraint that I proposed above, this also cannot happen by tampering with the db (unless on purpose and then this is on them)

We can remove `normalizedHeartbeatBarDays` and replace it with `heartbeatBarDays`